src/server/pachyderm_test.go

```diff
@@ -3799,6 +3805,78 @@ func TestPipelinePartialResourceRequest(t *testing.T) {
 	}, backoff.NewTestingBackOff()))
 }
 
+func TestPodSpecOpts(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+
+	c := getPachClient(t)
+	require.NoError(t, c.DeleteAll())
+	// create repos
+	dataRepo := tu.UniqueString("TestPodSpecOpts_data")
+	pipelineName := tu.UniqueString("TestPodSpecOpts")
+	require.NoError(t, c.CreateRepo(dataRepo))
+	priorityClassName := "system-cluster-critical"
+	// Resources are not yet in client.CreatePipeline() (we may add them later)
+	_, err := c.PpsAPIClient.CreatePipeline(
+		context.Background(),
+		&pps.CreatePipelineRequest{
+			Pipeline: &pps.Pipeline{pipelineName},
+			Transform: &pps.Transform{
+				Cmd: []string{"cp", path.Join("/pfs", dataRepo, "file"), "/pfs/out/file"},
+			},
+			ParallelismSpec: &pps.ParallelismSpec{
+				Constant: 1,
+			},
+			Input: &pps.Input{
+				Atom: &pps.AtomInput{
+					Repo:   dataRepo,
+					Branch: "master",
+					Glob:   "/*",
+				},
+			},
+			SchedulingSpec: &pps.SchedulingSpec{
+				NodeSelector: map[string]string{
+					"foo": "bar",
```

Comment:
Is the idea that this blocks the pipeline from scheduling? Or does this label get applied to the nodes somewhere? Or am I misunderstanding how this feature works?

Reply:
Yup, this blocks the pipeline from scheduling, but all the test does is wait for the pod to exist (it can exist without being scheduled) and then check to make sure that its pod spec reflects the values in the pipeline. I'll add a comment explaining this.